src/maker.rs

```diff
@@ -141,45 +173,55 @@ impl Maker {
                         position: Position::Buy,
                         ..
                     } => {
-                        // We are buying BTC for DAI
-                        // Given an order rate of: 1:9000
-                        // It is NOT profitable to buy, if the current rate is greater than the order rate.
-                        // 1:8800 -> We give less DAI for getting BTC -> Good.
-                        // 1:9200 -> We have to give more DAI for getting BTC -> Sucks.
-                        if order_rate > current_profitable_rate {
-                            return Ok(TakeRequestDecision::RateNotProfitable);
-                        }
-
-                        let updated_dai_reserved_funds =
-                            self.dai_reserved_funds.clone() + order.quote;
-                        if updated_dai_reserved_funds > self.dai_balance {
-                            return Ok(TakeRequestDecision::InsufficientFunds);
+                        match self.dai_balance.clone() {
+                            Some(dai_balance) => {
+                                // We are buying BTC for DAI
+                                // Given an order rate of: 1:9000
+                                // It is NOT profitable to buy, if the current rate is greater than the order rate.
+                                // 1:8800 -> We give less DAI for getting BTC -> Good.
+                                // 1:9200 -> We have to give more DAI for getting BTC -> Sucks.
+                                if order_rate > current_profitable_rate {
```

Comment:
```suggestion
                                if order_rate >= current_profitable_rate {
```
Otherwise it means we never accept orders we just created!

Reply:
Please add a test for this logic: take an order we just created.

Reply:
This is the **exit** path, it should be fine as is. 

I added one test for buy and one test for sell so it is more obvious:

```published_sell_order_can_be_taken```
```published_buy_order_can_be_taken```

Reply:
> This is the **exit** path, it should be fine as is.

I am not sure to understand this comment. The code seems wrong to me and no test seems to have caught that. Why is it fine?

Reply:
The code is not wrong. We exit early if the rate is not profitable.
If the rate is the same it is profitable.
I added 2 tests to show that it works as expected.